NewTutorials/Tutorial/TutorialCasesViewController.swift

```diff
@@ -0,0 +1,91 @@
+//  Copyright © 2017 Oath. All rights reserved.
+
+import UIKit
+import OneMobileSDK
+import PlayerControls
+
+func setupPlayingVideos(vc: TutorialCasesViewController) {
+    func onSuccess(player: Player) {
+        let playerViewController = PlayerViewController()
+        playerViewController.contentControlsViewController = DefaultControlsViewController()
+        playerViewController.player = player
+        vc.navigationController?.pushViewController(playerViewController, animated: true)
+    }
+    
+    vc.props = .init(
+        rows: [
+            TextCell.ViewModel.init(name: "Single video", action: {
+                vc.show(player: singleVideo().dispatch(on: .main).onSuccess(call: onSuccess))
+            }),
+            TextCell.ViewModel.init(name: "Array of videos", action: {
+                vc.show(player: arrayOfVideos().dispatch(on: .main).onSuccess(call: onSuccess))
+            }),
+            TextCell.ViewModel.init(name: "Video playlist", action: {
+                vc.show(player: videoPlaylist().dispatch(on: .main).onSuccess(call: onSuccess))
+            }),
+            TextCell.ViewModel.init(name: "Muted video", action: {
+                func mute(player: inout Player) { player.mute() }
+                vc.show(player: singleVideo().map(mute).dispatch(on: .main).onSuccess(call: onSuccess))
+            }),
+            TextCell.ViewModel.init(name: "Video without autoplay", action: {
+                vc.show(player: singleVideo(isAutoplay: false).dispatch(on: .main).onSuccess(call: onSuccess))
```

Comment:
I think action should operate on a `Player` object not on Future - you are subscribing to on success  in 2 places - when the only one needed.